### PR TITLE
Split --nodaemon option into --nodaemon and --verbose

### PR DESCRIPTION
**Pull request by schaal**
Don't change the log level when using --nodaemon. Add --verbose option to set the log level to debug.

This is for better integration with systemd to be able to prevent pilight from forking without increasing the log level.

### CODE REVIEW

**Comment by CurlyMoo** [comment]
All pilight core options are capatilized, so can you capatalize the `V` as well?

**Comment by schaal** [comment]
> All pilight core options are capatilized, so can you capatalize the `V` as well?

Turns out `V` is already taken by the version option. I choose `I` instead for the short option as in "debug **I**nformation" (as `D` is already taken as well).

**Comment by CurlyMoo** [comment]
What about just leaving the `I` out and keep the debuglevel option.

**Comment by CurlyMoo** [comment]
Maybe we should add a `--debug` option so the user will can have a `nodaemon` mode and `debuglevel` at once?

**Comment by schaal** [comment]
This now does the following:

* Don't set the log level to debug when nodaemon==1
* Don't override the log level set in the config file when nodaemon==1
* If any command line option is given that changes the verbosity (stacktracer, debuglevel, etc...) ignore the value in the config file.

**Comment by schaal** [comment]
> Maybe we should add a --debug option so the user will can have a nodaemon mode and debuglevel at once?

My preference would be to just have the two options independent of each other, but I can add the debug option if you want.

**Comment by CurlyMoo** [comment]
> My preference would be to just have the two options independent of each other, but I can add the debug option if you want.

Let's see what others think.

**Comment by pilino1234** [comment]
I quite like the current state of this PR as described here: https://github.com/pilight/pilight/pull/363#issuecomment-313909217

Although I have a couple of improvments:

- I don't like the name of the `--debuglevel` option currently, I think `verbosity` would be better.
- The possible debuglevels should be listed somewhere, maybe in the usage summary since `man` pages don't exist ([yet](https://github.com/pilight/pilight/pull/345)). A different alternative would be to use chaining options to change verbosity, so that `-vvvvv` would be more verbose than `-vv`, for example.
- Short options should be added for any of these that don't have them yet, because combining `--nodaemon`/`-D` with what `--debuglevel` does now is done quite commonly and should be easily accessible. Otherwise, adding a third option like what @CurlyMoo suggests (https://github.com/pilight/pilight/pull/363#issuecomment-313911447) should be done.

**Comment by CurlyMoo** [comment]
> I don't like the name of the --debuglevel option currently, I think verbosity would be better.

The `--debuglevel` is currently an option for developers which give information about the internals of pilight, which is much more verbose then the regular `LOG_*` verbosities.

> A different alternative would be to use chaining options to change verbosity, so that -vvvvv would be more verbose than -vv, for example.

This way of having stacked shortcodes are not supported (yet).

**Comment by pilino1234** [comment]
> The --debuglevel is currently an option for developers which give information about the internals of pilight, which is much more verbose then the regular LOG_* verbosities.

Whoops, that was a misread on my part. Basically, what I'm wondering is how do I achieve the same behaviour as what `pilight-daemon -D` did previously?

**Comment by CurlyMoo** [comment]
> Whoops, that was a misread on my part. Basically, what I'm wondering is how do I achieve the same behaviour as what pilight-daemon -D did previously?

That would require a `--verbosity` option set to `LOG_DEBUG` and the `-D` option combined.

**Comment by pilino1234** [comment]
That's what I'd like to have, yes.

**Comment by schaal** [comment]
I incorporated some of the feedback and it now looks like this:

* Keep the current behavior of the `-D` option (no forking and set log level to debug), but rename the long option to `--debug`.
* Add option `--nodaemon`/`-N` which prevents forking, but doesn't change the log level
* Command line options override the log level set in the config file

**Comment by CurlyMoo** [comment]
Maybe we can change `--nodaemon` to `--foreground` and `-F`. That's how it's called in other programs.

**Comment by schaal** [comment]
> Maybe we can change --nodaemon to --foreground and -F. That's how it's called in other programs.

Done

**Comment by pilino1234** [comment]
:+1:

File: daemon.c [hunk 1]
```diff
@@ -2329,6 +2336,19 @@ int start_pilight(int argc, char **argv) {
 
 	registerVersion();
 
+	// let verbosity level from command line take precedence over config file
+	if(!verbosity_changed) {
```
Comment (CurlyMoo) [inline daemon.c]: Can you make this `if(verbosity_changed == 0) {`

**Comment by CurlyMoo** [comment]
Can you squash everything into one commit?

**Comment by CurlyMoo** [comment]
Thanks!

**Pull request merged**